esphome/components/udp/udp_component.cpp

```diff
@@ -245,7 +245,7 @@ void UDPComponent::setup() {
     }
     struct sockaddr_in server {};
 
-    socklen_t sl = socket::set_sockaddr_any((struct sockaddr *) &server, sizeof(server), this->port_);
+    socklen_t sl = socket::set_sockaddr((struct sockaddr *) &server, sizeof(server), "0.0.0.0", this->port_);
```

Comment:
This just hard-codes the IPV4 broadcast address. What if you actually want to use IPV6? I didn't do any testing of the udp component with IPV6 and I'm sure it doesn't work right now, but it would be a better fix than this. Of course it may be impossible or too hard, in which case this change would be a reasonable work-around.
```suggestion
    server.sin_family = AF_INET;
    server.sin_addr.s_addr = ESPHOME_INADDR_ANY;
    server.sin_port = htons(this->port_);
```

Reply:
Yeah, I'll address IPv6 unicast and multicast in a separate PR, but for now I just like to be able to have it at least use IPv4 in a dual stack network.